elfo-core/src/supervisor.rs

```diff
@@ -323,26 +332,33 @@ where
                 Err(panic) => ActorStatus::FAILED.with_details(panic_to_string(panic)),
             };
 
-            let should_restart = {
-                let object = sv.objects.get(&key).expect("where is the current actor?");
-                let actor = object.as_actor().expect("a supervisor stores only actors");
-
-                let rp_override = actor.restart_policy();
-                let restart_policy = rp_override.as_ref().unwrap_or(&sv.restart_policy);
-                let should_restart = match restart_policy.mode {
-                    RestartMode::Always => true,
-                    RestartMode::OnFailures => new_status.is_failed(),
-                    RestartMode::Never => false,
-                };
-
-                actor.set_status(new_status);
-                should_restart
-            };
-
-            let need_to_restart = should_restart && !sv.control.read().stop_spawning;
-            if need_to_restart {
-                let after = backoff.next();
-
+            let object = sv.objects.get(&key).expect("where is the current actor?");
+            let actor = object.as_actor().expect("a supervisor stores only actors");
+
+            let rp_override = actor.restart_policy();
+            // Select the restart policy with the next priority:
+            //     ctx.set_restart_policy(..) > config > blueprint (sv.restart_policy).
```

Comment:
> sv.restart_policy

I think it should be named as `default_restart_policy` now to avoid confusion for readers

Reply:
It will be unexpected if I explicitly change the `restart_policy` via the `ctx` and regain it back when the configuration is reloaded.

Reply:
I renamed `sv.restart_policy` to `default_restart_policy` and introduced a new attribute, `group_restart_policy`, within `sv.control`. This attribute is updated during a `config_update` if specified in the configuration. If not explicitly defined, it defaults back to `default_restart_policy`. 

So in the code above we will chose either the actor `rp_override` or the `group_restart_policy`